app/models/user.rb

```diff
@@ -35,6 +37,18 @@ class User < ApplicationRecord
 
   scope :subscribed_to_daily_summary, -> { where(daily_summary: true) }
 
+  def active_for_authentication?
+    super && !discarded?
+  end
+
+  def can_be_discarded?
+    !cannot_be_discarded?
+  end
+
+  def cannot_be_discarded?
+    invoices.issued.present?
+  end
```

Comment:
What's the purpose of that? Why not combine into single method?

Reply:
Just semantic sugar. I'll change it to single, 'cause the main condition is 'if it can be discarded'

Reply:
Perhaps `discardable?` is more eloquent?

Reply:
Is there really a need to use gem for such simple feature?

Reply:
discardable? looks good for me.
As for gem - just trying not to reinvent a bike every time.